sycl/test/basic_tests/group_async_copy.cpp

```diff
@@ -0,0 +1,150 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.run
+// RUN: %GPU_RUN_PLACEHOLDER %t.run
+// RUN: %CPU_RUN_PLACEHOLDER %t.run
+// RUN: %ACC_RUN_PLACEHOLDER %t.run
+// RUN: env SYCL_DEVICE_FILTER=host %t.run
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <typeinfo>
+
+using namespace cl::sycl;
+
+template <typename T> class KernelName;
+
+// Define the number of work items to enqueue.
+const size_t NElems = 32;
+const size_t WorkGroupSize = 8;
+const size_t NWorkGroups = NElems / WorkGroupSize;
+
+template <typename T>
+void initInputBuffer(buffer<T, 1> &Buf, size_t Stride = 0) {
+  auto Acc = Buf.template get_access<access::mode::write>();
+  for (size_t I = 0; I < Buf.get_count(); I += WorkGroupSize) {
+    for (size_t J = 0; J < WorkGroupSize; J++)
+      Acc[I + J] = I + J + ((J % Stride == 0) ? 100 : 0);
+  }
+}
+
+template <typename T> void initOutputBuffer(buffer<T, 1> &Buf) {
+  auto Acc = Buf.template get_access<access::mode::write>();
+  for (size_t I = 0; I < Buf.get_count(); I++)
+    Acc[I] = 0;
+}
+
+template <typename T> struct is_vec : std::false_type {};
+template <typename T, size_t N> struct is_vec<vec<T, N>> : std::true_type {};
+
+template <typename T> bool checkEqual(vec<T, 1> A, size_t B) {
+  T TB = B;
+  return A.s0() == TB;
+}
+
+template <typename T> bool checkEqual(vec<T, 4> A, size_t B) {
+  T TB = B;
+  return A.x() == TB && A.y() == TB && A.z() == TB && A.w() == TB;
+}
+
+template <typename T>
+typename std::enable_if<!is_vec<T>::value, bool>::type checkEqual(T A,
+                                                                  size_t B) {
+  T TB = B;
+  return A == TB;
+}
+
+template <typename T> std::string toString(vec<T, 1> A) {
+  std::string R("(");
+  return R + std::to_string(A.s0()) + ")";
+}
+
+template <typename T> std::string toString(vec<T, 4> A) {
+  std::string R("(");
+  R += std::to_string(A.x()) + "," + std::to_string(A.y()) + "," +
+       std::to_string(A.z()) + "," + std::to_string(A.w()) + ")";
+  return R;
+}
+
+template <typename T = void>
+typename std::enable_if<!is_vec<T>::value, std::string>::type toString(T A) {
+  return std::to_string(A);
+}
+
+template <typename T> int checkResults(buffer<T, 1> &OutBuf, size_t Stride) {
+  auto Out = OutBuf.template get_access<access::mode::read>();
+  int EarlyFailout = 20;
```

Comment:
How did we get 20 here? Does early failout helps in any way here i.e. in testing?
Is 20 an arbitrary number less than `NElems`?

Reply:
20 is an arbitrary number, it is reasonable big enough to let the error log observer see the pattern and have at least some initial guess on what went wrong with the copy even before running the test locally.